workflow/src/text_chat.py

```diff
@@ -287,8 +290,72 @@ def make_chat_request(
     return prompt, response
 
 
+@time_it
+def make_litellm_chat_request(
+    prompt: str,
+    model: str,
+    temperature: float,
+    max_tokens: Optional[int],
+    top_p: int,
+    frequency_penalty: float,
+    presence_penalty: float,
+) -> Tuple[str, str]:
+    """Sends a chat request to OpenAI/Azure/Cohere/Anthropic/Bard/Pal model and returns the prompt and
+    response as a tuple.
+
+    Args:
+        prompt (str): The prompt to send to the model.
+        model (str): The model to be used, see https://litellm.readthedocs.io/en/latest/supported/ for models
+        temperature (float): Controls the "creativity" of the response. Higher values result in more diverse responses.
+        max_tokens (Optional[int]): The maximum number of tokens (words) in the response.
+        top_p (int): Controls the "quality" of the response. Higher values result in more coherent responses.
+        frequency_penalty (float): Controls the model's tendency to repeat itself.
+        presence_penalty (float): Controls the model's tendency to stay on topic.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the prompt and the response from the model.
+    """
+    intercept_custom_prompts(prompt)
+    prompt = prompt_for_alias(prompt)
+    messages = create_message(prompt)
+    write_to_cache("last_chat_request_successful", True)
+
+    try:
+        response = (
+            litellm.completion(
+                model=model,
+                messages=messages,
+                temperature=temperature,
+                max_tokens=max_tokens,
+                top_p=top_p,
+                frequency_penalty=frequency_penalty,
+                presence_penalty=presence_penalty,
+            )
+            .choices[0]
+            .message["content"]
+        )
+
+    except Exception as exception:  # pylint: disable=broad-except
+        response = exception_response(exception)
+        write_to_cache("last_chat_request_successful", False)
+        log_error_if_needed(
+            model=__model,
+            error_message=exception._message,  # type: ignore  # pylint: disable=protected-access
+            user_prompt=prompt,
+            parameters={
+                "temperature": temperature,
+                "max_tokens": max_tokens,
+                "top_p": top_p,
+                "frequency_penalty": frequency_penalty,
+                "presence_penalty": presence_penalty,
+            },
+        )
+
+    return prompt, response
+
+
 exit_on_error()
-__prompt, __response = make_chat_request(
+__prompt, __response = make_litellm_chat_request(
```

Comment:
Hey! Nice job! Just wondering. How is the `model (str)` parameter passed to the `make_litellm_chat_request` function?

Reply:
Checking the PR by @ishaan-jaff model(str) should be one of the supported models at https://litellm.readthedocs.io/en/latest/supported/

In the current configuration, there is a dropdown model called ChatGPT model that has 'ChatGPT', 'GPT-4' and 'GPT-4 (32k)'. IMHO, this dropdown could be renamed 'GPT Model' (more recognizable than LLM Model) and include the strings for the supported models.

To me the transition as seamless as possible.